### DISCUSSION TITLE
Is `ulid` really necessary, wouldnt uuid:v7 do the job?

### DISCUSSION THREAD

**Discussion by kirillsalykin** [maxcountryman/underway#20] [Q&A]
It seems that both `ulid` and `uuid` crates are used, so it makes me wonder, maybe just having uuid:v7 would be enough?

**Comment by maxcountryman** [comment]
UUIDv7 is probably fine too. I'm not sure how much difference it makes in practice. Last I looked the uuid crate did not support v7 (maybe it does now?)

**Reply by kirillsalykin** [reply]
AFAIK, not much different, they both sortable because first part is timestamp.
v7 suppose to be officially supported (it is not under unstable anymore)

**Reply by maxcountryman** [reply]
Technically ULID is more resistant to sub second collisions because it doesn't have to use bits on versioning etc.

I don't feel strongly one way or the other, so if it's available I'm okay to move to UUIDv7.

**Reply by victoronascimento** [reply]
Just to add to the topic, some bits to add here in favor of UUIDv7:

- pg_partman 5.2+ supports [partitioning by range with UUIDv7](https://github.com/pgpartman/pg_partman/blob/development/doc/pg_partman_howto.md#simple-time-based-with-uuidv7-type-1-partition-per-day) (no need for composite PKs). Might be used on setups that want to leave jobs lingering longer and not hurt active jobs performance;
- UUIDv7 is accepted and to be included on the [next postgres release 18](https://www.postgresql.org/about/news/postgresql-18-beta-1-released-3070/)
